tempy/elements.py

```diff
@@ -385,6 +398,42 @@ def dump(self, filename, **kwargs):
             file_to_write.write(self.render(**kwargs))
             self._template = '<style>{css}</style>'
 
+    def replace_element(self, selector_list, new_style, ignore_error=True):
+        if new_style is None or not isinstance(new_style, (str, dict)) or not new_style:
+            if ignore_error:
+                return
+            else:
+                raise WrongArgsError(self, new_style, 'Second argument should be a non-empty string or dictionary.')
+
+        try:
+            element_node = self.find_attr(selector_list)
+        except (AttrNotFoundError, WrongArgsError) as wrong_args_error:
+            if ignore_error:
+                return
+            else:
+                print(wrong_args_error.__repr__())
+                raise
+
+        if element_node:
+            element_node[selector_list[-1]] = new_style
+        elif not element_node and selector_list[0] in self.attrs['css_attrs']:
+            self.attrs['css_attrs'] = new_style
```

Comment:
Is it possible that element_node is False/None/Empty?

Reply:
This function returns the parent_node, so you can actually set the element node as shown. It is None  if there is no parent present, which indicates that that is the root css attribute node. In this example
```
{'html': {
            'body': {
                'color': 'red',
                Div: {
                    'color': 'green',
                    'border': '1px'
                },
               link: {'color': 'yellow'}
            }
        },
        '#myid': {'color': 'blue'}
    }
}
```
html has no parent, thus replacing it will fall into this branch.

Reply:
using your example, the line `element_node = self.find_attr(['html', ])` is equivalent to 
```
element_node = {'html': {'body': {'color': 'red', 'Div': {'color': 'green', 'border': '1px'}, 'link': {'color': 'yellow'}}}, '#myid': {'color': 'blue'}}
```

so `bool(element_node)` will always be `True`. In other words, if the path is valid, there is always  parent node of the path...

Reply:
No, it will return the parent_node which is None in this case, since this node has no parent. The method find_attr returns the parent node of the found node. If the node is not found it will raise AttrNotFoundError in every case, else it returns the parent_node, which is None for this case.

Reply:
Ok, was just curious for it seemed a way to replace the whole `self.attrs['css_attrs']` dict.. ;)
Merging now, thanks again!

Reply:
No problem! I'll add the clear method next if that's ok